### PR TITLE
fix: handle duplicate sources

### PR DESCRIPTION
**Pull request by rexagod**
## Description

Skip appending sources with the same key for ConfigMaps and Secrets in the AlertmanagerConfiguration.

Fixes: #5905

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [x] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

### CODE REVIEW

File: pkg/alertmanager/statefulset.go [hunk 1]
```diff
@@ -531,8 +534,14 @@ func makeStatefulSetSpec(a *monitoringv1.Alertmanager, config Config, tlsAssetSe
 	amCfg := a.Spec.AlertmanagerConfiguration
 	if amCfg != nil && len(amCfg.Templates) > 0 {
 		sources := []v1.VolumeProjection{}
+		configmapKeys := sets.Set[string]{}
+		secretKeys := sets.Set[string]{}
```
Comment (ArthurSens) [inline pkg/alertmanager/statefulset.go:538]: If a secret and a configmap have the same name, I believe we'll also have conflicting paths, is that correct?

Don't we need to use the same set for both?

**Review by ArthurSens — COMMENTED** [review]
Thanks for the PR!

Could we also have a unit test to make sure we don't mount duplicated paths?

File: pkg/alertmanager/statefulset.go (hunk 1 above)
Comment (rexagod) [reply pkg/alertmanager/statefulset.go]: Hmm, that makes sense. Patched!

**Review by ArthurSens — APPROVED** [review]
Thanks!

**Pull request merged**